pep-0690.rst

```diff
@@ -268,31 +286,26 @@ Lazy imports are represented internally by a "lazy import" object. When a lazy
 import occurs (say ``import foo`` or ``from foo import bar``), the key ``"foo"``
 or ``"bar"`` is immediately added to the module namespace dictionary, but with
 its value set to an internal-only "lazy import" object that preserves all the
-necessary metadata to execute the import later. The ``DictKeysKind`` for the
-module namespace dictionary is updated from e.g. ``DICT_KEYS_UNICODE`` to
-``DICT_KEYS_UNICODE_LAZY`` to signal that this particular dictionary may contain
-lazy import objects.
-
-(In case someone adds a non-unicode key to a module namespace dictionary also
-containing lazy import objects, e.g. via ``globals()[42] = "foo"``, there is
-also ``DICT_KEYS_GENERAL_LAZY``, but in most cases this is not needed.)
-
-Anytime a key is looked up in a dictionary with ``DICT_KEYS_UNICODE_LAZY`` or
-``DICT_KEYS_GENERAL_LAZY``, the value is checked to see if it is a lazy import
-object. If so, the import is immediately executed, the lazy import object is
-replaced in the dictionary by the actual imported value, and the imported value
-is returned from the lookup.
+necessary metadata to execute the import later. A new boolean flag in
+``PyDictKeysObject`` (``dk_lazy_imports``) is set to signal that this particular
+dictionary may contain lazy import objects.
+
+Anytime a key is looked up in a dictionary to extract its value, the
```

Comment:
In these cases It checks the type of the value that is extracted from the dictionary (to see if it's laly), since it's a type check only it does not also check the `dk_lazy_imports` flag.